metal/classifier.py

```diff
@@ -182,8 +212,28 @@ def _train_model(self, train_data, loss_fn, dev_data=None, log_writer=None):
                 train_config["checkpoint_config"]
             )
 
+        # which iteration to start with
+        start_iteration = 0
+
+        if restore_state:
```

Comment:
Can we move this loading functionality to helper? _train_model() is already uncomfortably long.